strawberry/types/fields/resolver.py

```diff
@@ -0,0 +1,85 @@
+import inspect
+from inspect import iscoroutinefunction
+from typing import Callable, Generic, List, Optional, Type, TypeVar
+
+from cached_property import cached_property  # type: ignore
+
+from strawberry.arguments import get_arguments_from_annotations
+from strawberry.exceptions import MissingArgumentsAnnotationsError
+from strawberry.types.types import ArgumentDefinition
+from strawberry.utils.inspect import get_func_args
+
+
+T = TypeVar("T")
+
+
+class StrawberryResolver(Generic[T]):
+    def __init__(self, func: Callable[..., T], *, description: Optional[str] = None):
+        self.wrapped_func = func
+        self._description = description
+
+    # TODO: Use this when doing the actual resolving? How to deal with async resolvers?
+    def __call__(self, *args, **kwargs) -> T:
+        return self.wrapped_func(*args, **kwargs)
+
+    # TODO: Return StrawberryArguments instead. Maybe this should be a classmethod
+    #       there instead?
+    # TODO: Return a set instead
+    @cached_property
+    def arguments(self) -> List[ArgumentDefinition]:
+        # TODO: Move to StrawberryArgument? StrawberryResolver ClassVar?
+        SPECIAL_ARGS = {"root", "self", "info"}
```

Comment:
I figured I'd turn this into a constant, but I wasn't quite sure where to put it. Not sure if it would be better within `StrawberryResolver` or the future `StrawberryArgument`